hudi-common/src/main/java/org/apache/hudi/common/model/OverwriteWithLatestAvroPayload.java

```diff
@@ -94,6 +94,6 @@ protected boolean isDeleteRecord(GenericRecord genericRecord) {
    * Return true if value equals defaultValue otherwise false.
    */
   public Boolean overwriteField(Object value, Object defaultValue) {
-    return defaultValue == null ? value == null : defaultValue.toString().equals(String.valueOf(value));
```

Comment:
Thanks for the fix. Can you please write a test case for few diff data types. In general, its a good practice to always accompany bugs with tests.

Reply:
Thanks for reviewing. I add an array type to data schema in the unit test.